### ISSUE TITLE
`consider-using-f-string` should not be issued with py-version < 3.6

### ISSUE THREAD

**Issue by scop** [PyCQA/pylint#5019]
### Bug description

f-strings were introduced in Python 3.6, they should not be recommended with earlier py-versions.

### Command used

```shell
echo '"{}".format("foo")' | pylint --py-version=3.5 --from-stdin mymodule | grep f-string
```

### Pylint output

```shell
mymodule:1:0: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)
```

### Expected behavior

No `consider-using-f-string` message when `py-version` is set to earlier than 3.6.

### Pylint version

```shell
pylint 2.11.0
astroid 2.8.0
Python 3.9.3 (default, Apr  3 2021, 00:13:19) 
[GCC 9.3.0]
```

**Comment by Pierre-Sassoulas** [comment]
Hmm, I don't think pylint is supposed to be able to analyze version < 3.6. The py-version is kinda misleading because it feels like it should work but there was clearly an assumption that python interpreter used == py-version, originally. What do you think @cdce8p  ?

**Comment by scop** [comment]
Hm, not sure what you mean by that assumption. But FWIW the way it's put in the release notes:
> Added py-version config key (if [MASTER] section). Used for version dependant checks. Will default to whatever Python version pylint is executed with.
...gives a very strong impression that it actually does work. In my opinion it definitely should :)

BTW the features entry for this option at https://docs.pylint.org/en/latest/technical_reference/features.html?highlight=py-version#general-options is quite confusing:

> Min Python version to use for version dependend checks. Will default to the version used to run pylint.
> Default: 3.8

It first says default is the version used to run pylint, then it says default is 3.8.
(While at it, I'd suggest rephrasing Min -> Minimum, and fixing dependend -> dependent. Would submit a PR for this but perhaps there will be bigger changes to the doc if I'm not right about what the option is for.)

**Comment by cdce8p** [comment]
> Hmm, I don't think pylint is supposed to be able to analyze version < 3.6. The py-version is kinda misleading because it feels like it should work but there was clearly an assumption that python interpreter used == py-version, originally.

That was my initial though as well. However, it does make sense to extend it to other checks. If I remember correctly we discussed that shortly that users might want this setting to work for other checks as well. I haven't though about `consider-using-f-string` as I've disabled it for me personally, but that would indeed be a good fit. It shouldn't be too difficult either. Just the question which other checks need to support it as well. I guess new issues is one way to find out, even if unplanned.

FYI: At the moment, `py-version` is only used in the `Typing` and `CodeStyle` extensions.

> But FWIW the way it's put in the release notes: ... ...gives a very strong impression that it actually does work. In my opinion it definitely should :)

I agree. That should have been put better into context.

> It first says default is the version used to run pylint, then it says default is 3.8.

The docs are autogenerated with Python 3.8. A bit unfortunate if I'm being honest
https://github.com/PyCQA/pylint/blob/da36529a6ae1fb83417e600528692530a27aac1f/pylint/lint/pylinter.py#L475-L484

> (While at it, I'd suggest rephrasing Min -> Minimum, and fixing dependend -> dependent. Would submit a PR for this but perhaps there will be bigger changes to the doc if I'm not right about what the option is for.)

Thanks for the suggestion! A PR would be welcome.

**Comment by cdce8p** [comment]
Opened #5024 that should address the main concern with the `consider-using-f-string` checker.
